spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -124,7 +124,7 @@ class CassandraTableScanRDD[R] private[connector](
 
   override def getPreferredLocations(split: Partition): Seq[String] =
     split.asInstanceOf[CassandraPartition]
-      .endpoints.map(_.getHostName).toSeq
+      .endpoints.flatMap(inet => Seq(inet.getHostName, inet.getHostAddress)).toSeq.distinct
```

Comment:
I see what you are doing now, a seq of both for all and flattening, I though you were wanting tuples